From 675a9bee6bff7622f22adc2adbe2c13579002f09 Mon Sep 17 00:00:00 2001
From: Shaun Tancheff <shaun@tancheff.com>
Date: Wed, 9 Mar 2016 22:53:29 -0600
Subject: [PATCH 18/25] Suggest: Use i_ino when stream_id is not set.

When no stream_id was set by user space, fall back to an i_ino hash.

Signed-off-by: Shaun Tancheff <shaun.tancheff@seagate.com>
---
 include/linux/fs.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index 3092f82..8eb31f4 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -682,9 +682,11 @@ struct inode {
 
 static inline unsigned int inode_streamid(struct inode *inode)
 {
-	if (inode)
-		return inode->i_streamid;
-
+	if (inode) {
+		if (inode->i_streamid)
+			return inode->i_streamid;
+		return ((inode->i_ino >> 16) ^ inode->i_ino) & 0xFFFF;
+	}
 	return 0;
 }
 
-- 
2.7.0

